### ISSUE TITLE
Save to file system it's taking too looooong.

### ISSUE THREAD

**Issue by psoaresbj** [prismicio/slice-machine#1020]
### Versions

- slice-machine-ui: v1.1.1
- node: v18.12.1

### Reproduction
When I simply press save to file system, that action takes a really LOOOOONG time (more than 5minutes).
I don't know if it's because:
- The project have more than 100 `customTypes` with a lot of fields in;
- A `prismic-types.d.ts` file is generated (file that I need to delete, since the project don't have TS implemented and linter requests to create a `tsconfig` file). As today, that file have 37.680 lines.

### What is expected?
Since it's a no `ts` project, should only update the *.json files... (expecting that's what's causing some kind of issue)...
We did a big jump on versions TBH (from `0.5.2` to `1.1.0`)

### Notes
With other projects (way smaller than this specific example: tech company with > 200 ppl team), there's no issue but stills generate TS file.

**Comment by angeloashmore** [comment]
Hi @psoaresbj, thanks for the report. The long time is likely due to `prismicio-types.d.ts`'s generation and is something we plan to address. The underlying library that generates types (`prismic-ts-codegen`) works well enough for small to medium sized projects, but it can be much too slow on large projects like yours.

We will post again here when we have an update to share. Thanks again for the report. 🙂

**Comment by psoaresbj** [comment]
Thanks for your feedback @angeloashmore.
It's really hard to do changes in customTypes and slices... Meanwhile, is there any hard option to skip types generation - we don't need it? 

Thanks in advance

**Comment by angeloashmore** [comment]
@psoaresbj Unfortunately, no, there is no way to disable type generation currently. We feel strongly about always generating types since it improves the development experience with no extra user effort. I understand how the slow performance can be a blocker, however.

A much faster type generation implementation is in the works in https://github.com/prismicio/prismic-ts-codegen/pull/51, which should be finished and published soon.

Until it is published, you could manually disable type generation by temporarily editing files in your `node_modules/@slicemachine/adapter-next/dist`. You can delete the following lines, which effectively disables type generation (`prismicio-types.d.ts` will still be generated).

```diff
// node_modules/@slicemachine/adapter-next/dist/lib/upsertGlobalContentTypes.js

  let contents = generateTypes({
-   customTypeModels,
-   sharedSliceModels,
-   clientIntegration: {
-     includeCreateClientInterface: true,
-     includeContentNamespace: true
-   },
    typesProvider: await detectTypesProvider({ cwd: project.root })
  });
```

If you want to persist this patch until https://github.com/prismicio/prismic-ts-codegen/pull/51 is merged, check out https://github.com/ds300/patch-package. Thanks for your patience!

**Comment by psoaresbj** [comment]
Thanks.
We're already doing that :)

**Comment by xrutayisire** [comment]
Hi,

We just released a new Slice Machine version 1.5.2 with adapter 0.3.5 🚀
It's now much faster 💨

Thank you for your patience and don't hesitate to reopen the issue if there is a problem.
